src/blocked.c

```diff
@@ -564,7 +565,10 @@ static void handleClientsBlockedOnKey(readyList *rl) {
         listIter li;
         listRewind(clients,&li);
 
-        while((ln = listNext(&li))) {
+        /* Only do count loops to avoid getting stuck in an endless loop. Because
+         * XREADGROUP with ">" may repeatedly reprocess command and blockForKeys. */
```

Comment:
```suggestion
        /* Avoid processing more than the initial count so that we're not stuck
           in an endless loop in case the reprocessing of the command blocks again. */
```